libr/bin/p/bin_dis.c

```diff
@@ -0,0 +1,302 @@
+/* radare2 - MIT - Copyright 2023 - keegan */
+
+#define R_LOG_ORIGIN "bin.dis"
+
+#include <r_types.h>
+#include <r_util.h>
+#include <r_lib.h>
+#include <r_bin.h>
+
+#include "../../arch/p/dis/dis.h"
+#include "../../arch/p/dis/dis.c"
+
+static bool check_buffer(RBinFile *bf, RBuffer *buf) {
+	ut64 pos = r_buf_tell (buf);
+	r_buf_seek (buf, 0, R_BUF_SET);
+	st32 magic;
+	if (!dis_read_operand (buf, &magic)) {
+		return false;
+	}
+	r_buf_seek (buf, pos, R_BUF_SET);
+	return magic == XMAGIC || magic == SMAGIC;
+}
+
+static bool load_buffer(RBinFile *bf, void **bin_obj, RBuffer *buf, ut64 loadaddr, Sdb *sdb) {
+	ut32 i;
+
+	if (!check_buffer (bf, buf)) {
+		return false;
+	}
+
+	RBinDisObj *o = R_NEW0 (RBinDisObj);
+	o->pcs = ht_uu_new0 ();
+
+	if (!dis_read_operand (buf, &o->header.magic)) {
+		goto invalid;
+	}
+	// skip signature, if needed
+	if (o->header.magic == SMAGIC) {
+		st32 signature_size;
+		if (!dis_read_operand (buf, &signature_size)) {
+			goto invalid;
+		}
+		if (signature_size < 0) {
+			goto invalid;
+		}
+		r_buf_seek (buf, signature_size, R_BUF_CUR);
+	}
+
+	if (!dis_read_operand (buf, &o->header.runtime_flags)) {
+		goto invalid;
+	}
+
+	if (!dis_read_operand (buf, &o->header.stack_extent)) {
+		goto invalid;
+	}
+	if (o->header.stack_extent < 0) {
+		goto invalid;
+	}
+
+	if (!dis_read_operand (buf, &o->header.code_size)) {
+		goto invalid;
+	}
+	if (o->header.code_size < 0) {
```

Comment:
```suggestion
	if (o->header.code_size < 1) {
```

if codesize is 0 it should be invalid too